### PR TITLE
Tms/merchants endpoint #1

### LINKED ISSUE #1: Merchants record endpoint

### PR DESCRIPTION
**Pull request by tmikeschu**
Worked with @Dpalazzari 

-Adds API endpoints for merchants: index, show, find, find_all, and random.

Needs in-depth review.

@sallymacnicholas we are having problems figuring out how to handle a string time parameter for finding merchants by created/updated_at. We've looked into this problem and can't figure it out. We've tried making a Time object from the string, and AR reads does not read them as the same. Thoughts?

Closes #1

### CODE REVIEW

**Comment by Dpalazzari** [comment]
The functionality for merchants works, barring a few set backs. Those will be ironed out with conversation with teachers. Can I merge @tmikeschu ?

File: config/routes.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
 Rails.application.routes.draw do
   namespace :api, defaults: { format: :json } do
     namespace :v1 do
-      get '/merchants/find', to: 'merchants_finder#show', as: 'merchant_finder'
-      get '/merchants/find_all', to: 'merchants_finder#index', as: 'merchants_finder'
-      get '/merchants/random', to: 'random_merchant#show', as: 'random_merchant'
+      namespace :merchants do
+        get '/find', to: 'search#show'
+        get '/find_all', to: 'search#index'
+        get '/random', to: 'random#show'
```
Comment (Dpalazzari) [inline config/routes.rb:7]: I like this. I think it's better and cleaner. @tmikeschu  👍 🚀

**Comment by tmikeschu** [comment]
@Dpalazzari , I agree that this basically works, but I'd feel more comfortable having a passing test for finding records by created/updated_at before merging. Thoughts?

File: config/routes.rb (hunk 1 above)
Comment (tmikeschu) [reply config/routes.rb:7]: @Dpalazzari your idea! Sorry we didn't run with it to begin with.

**Comment by Dpalazzari** [comment]
I agree. to be determined with conversation with @jmejia and @sallymacnicholas

@tmikeschu I might need this merged do begin work tomorrow morning though. Can we make a waffle card for the created?update _at tests so we can merge without forgetting to come back to these errors?

**Comment by tmikeschu** [comment]
@Dpalazzari I noticed the same thing! Check the new PR that will allow testing for new endpoints. I'm going to follow the folder and namespacing in this branch structure as I work on a new one.

**Comment by tmikeschu** [comment]
@Dpalazzari : To review what we talked about: 

- Changed the name type to be 'citext', which is case-ignored by default.

- Removed the timestamp attributes and added them again (was unable to `change_column`) with `precision: 0` (see code). This makes it so the postgres timestamp and the json-ized timestamp are read as the same.

- Added request helper and refactored tests with a verify attributes method.

- Added and updated tests that involve case-insensitivity and timestamps.

Ready for merge!

@sallymacnicholas

**Comment by Dpalazzari** [comment]
Looks good @tmikeschu I will merge. 

FYI @jmejia @sallymacnicholas @case-eee

**Pull request merged**